pkg/k8s/pod.go

```diff
@@ -442,11 +467,25 @@ func buildInitContainerVolumes(name string, volumes []*Volume, files []*File) []
 		return []v1.VolumeMount{} // return empty slice if no volumes are specified
 	}
 
-	containerVolumes := []v1.VolumeMount{
-		{
+	var containerVolumes []v1.VolumeMount
+	if len(volumes) != 0 {
```

Comment:
It feels like we should have some context captured in comments as to the split out of these if conditions. 

I don't understand why we are checking `len(volumes)` but not doing anything with it in either if condition.

Reply:
What do you mean by not doing anything?

But I added some comments to explain why we need it like that.

See: https://github.com/celestiaorg/knuu/pull/589/commits/08c587ebe34f72819484a0c80df66a6165baaa90